concordium-contracts-common/src/impls.rs

```diff
@@ -281,6 +282,95 @@ impl Deserial for AccountBalance {
     }
 }
 
+impl<Kind> Serial for Threshold<Kind> {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> { self.threshold.serial(out) }
+}
+
+impl<Kind> Deserial for Threshold<Kind> {
+    fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
+        let threshold: u8 = u8::deserial(source)?;
+        if threshold == 0 {
+            Err(ParseError {})
+        } else {
+            Ok(Self {
+                threshold,
+                kind: marker::PhantomData,
+            })
+        }
+    }
+}
+
+impl<Kind> TryFrom<u8> for Threshold<Kind> {
+    type Error = ZeroSignatureThreshold;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        if value == 0 {
+            Err(ZeroSignatureThreshold)
```

Comment:
This type has a generic `Threshold` name (meaning people could think to use it for all kinds of thresholds), but since it returns a very specific error when the value is set to 0 (referencing signatures), the `threshold` is somewhat meant for a specific use case. 

Suggestion: Would it make it clearer if we call it `SignatureThreshold` instead of `Threshold`?

I assume you called it just `Threshold` because it is difficult to pick to either call it `SignatureThreshold` or `AccountThreshold`.

Reply:
Ok, Renamed to `NonZeroThresholdU8`.